diff --git a/scripts/build-keccak-code-package.sh b/scripts/build-keccak-code-package.sh
index c225aa4..08ef01c 100755
--- a/scripts/build-keccak-code-package.sh
+++ b/scripts/build-keccak-code-package.sh
@@ -10,7 +10,7 @@ then
 	mkdir -p vendor
 	if [ ! -f "vendor/XKCP-master.zip" ];
 	then
-		wget -O vendor/XKCP-master.zip https://github.com/XKCP/XKCP/archive/master.zip
+		curl -L https://github.com/XKCP/XKCP/archive/master.zip -o vendor/XKCP-master.zip
 	fi
 
 	cd vendor
